packages/react/src/hooks/useNotifications.ts

```diff
@@ -66,14 +76,17 @@ export const useNotifications = (
     setErrorNextPage(null);
     setIsLoadingNextPage(true);
 
-    return new Promise(async (resolve, reject) => {
+    return new Promise(async (resolve) => {
+      // wait for the next page function to be ready
       if (!nextPage) {
-        const err = new Error("No more pages to fetch");
-        setErrorNextPage(err.message);
-        return reject(err);
+	await waitFor(() => Boolean(nextPage))
```

Comment:
Because `nextPage` is only undefined while it's loading, not when there are "no more pages to fetch". So it unnecessarily throws. Now, when calling `fetchNextPage` while `nextPage` is loading up, it'll just wait for it to be ready

Reply:
Understood, cool solution, thanks!